ddtrace/settings/config.py

```diff
@@ -35,13 +37,64 @@ def _deepmerge(source, destination):
     return destination
 
 
+def get_error_ranges(error_range_str):
+    error_ranges = []
+    error_range_str = error_range_str.strip()
+    error_ranges_str = error_range_str.split(",")
+    for error_range in error_ranges_str:
+        values = error_range.split("-")
+        try:
+            values = [int(v) for v in values]
+        except ValueError:
+            log.exception("Error status codes was not a number %s", values)
+            continue
+        error_range = [min(values), max(values)]
+        error_ranges.append(error_range)
+    return error_ranges
+
+
 class Config(object):
     """Configuration object that exposes an API to set and retrieve
     global settings for each integration. All integrations must use
     this instance to register their defaults, so that they're public
     available and can be updated by users.
     """
 
+    class HTTPServerConfig(object):
```

Comment:
👍🏻 I do like this a bit better, adding caching to the other place just felt like we were using caching to compensate for a poor implementation.

Reply:
can we add type hinting to everything in `HTTPServerConfig` and `get_error_ranges`?

can this be an `attr` class?